server/service/queries.go

```diff
@@ -516,6 +534,23 @@ func (svc *Service) ApplyQuerySpecs(ctx context.Context, specs []*fleet.QuerySpe
 				Message: fmt.Sprintf("query payload verification: %s", err),
 			})
 		}
+
+		dbQuery, err := svc.ds.QueryByName(ctx, query.TeamID, query.Name)
+		if err != nil && !fleet.IsNotFound(ctxerr.Unwrap(err)) {
+			return ctxerr.Wrap(ctx, fleet.NewBadGatewayError(fmt.Sprintf("fetching saved query"), err))
```

Comment:
Not sure bad gateway is the proper response code.

When datastore methods fail with unexpected errors we just return `ctxerr.Wrap(ctx, err, "fetching saved query")`. Our transport layer then returns a 500 (`server/service/transport.go`).

Reply:
Gotcha, will remember that going forward!